source/agora/node/Registry.d

```diff
@@ -823,14 +825,28 @@ private struct ZoneData
             "FOREIGN KEY(pubkey) REFERENCES registry_%s_utxo(pubkey) ON DELETE CASCADE, " ~
             "PRIMARY KEY(pubkey, address))", zone_name, zone_name);
 
+        // Initialize common fields
         this.db.execute(query_sig_create);
         this.db.execute(query_addr_create);
 
+        this.soa = this.config.fromConfig(this.root);
+        this.updateSOA();
+    }
+
+    /***************************************************************************
+
+        Update the SOA RR of the zone
+
+        Zone serial is set to current time when the zone is primary.
+
+    ***************************************************************************/
+
+    public void updateSOA () @trusted
+    {
         if (this.type == ZoneType.primary)
         {
-            // Serial's value wraps around so the cast is safe
-            const serial = cast(uint) Clock.currTime(UTC()).toUnixTime();
-            this.soa = this.config.fromConfig(this.root, serial);
+            this.soa.serial = cast(uint) Clock.currTime(UTC()).toUnixTime();
+            return;
```

Comment:
```D
if (this.type != ZoneType.primary)
    return;
this.soa.serial = ...;
```

The `else` is not required anymore since we're no longer in the constructor.